src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let index = 0; index < this.length; index++) {
```

Comment:
usually we use i in for-loops

```suggestion
    for (let i = 0; i < this.length; i++) {
```